# 2021-12-01 Wednesday

(mashr) 10:20 zyqi@login02:/data/cotton/zyqi/ma_2021_ng/cotton401vsZXL.vcf
$zcat cotton401vsZXL.vcf.gz|awk '/^#C/{print NF}'
410




bss "fastq-dump --gzip --split-3 test.sra"
test_1.fastq.gz
序列的名称也为test开头，test.1 test.2 ...


## 下载测序数据
/data/cotton/zyqi/ma_2021_ng/SRR_2
for i in {00..23};do bss -q smp  "sh download_$i";done
同时下载的数量到了956就上不去了？




共定位主要针对那些位于非编码区的GWAS位点，假设这个位点影响了某个基因的表达从而发挥功能。

Identifying whether or not the same variant is causal in both GWASs and expression quantitative trail locus (eQTL) studies is challenging because of the uncertainty induced by linkage disequilibrium and the fact that some loci harbor multiple causal variants.[@hormozdiari2016] #共定位





# 2021-12-02 Thursday

/data/cotton/zyqi/gwas/phe_349
一个让人疑惑的目录

王老师做GWAS的目录
/data/cotton/MaojunWang/WMJ_fiberFullPopulationRNAseq/Trait_GWAS


#EMMAX 使用
UHML_EMMAX_hBN.ps被覆盖
```shell
cat UHML_EMMAX_hIBS_cov.ps|awk -F'[\t_]' '{print $1"_"$2,$1,$2,$4}' OFS='\t'|sed '1iSNP\tCHR\tBP\tP' > blup_UHML_hIBS_cov.manhattan.txt &

Rscript qqman.script2.R blup_UHML_hIBS_cov.manhattan.txt blup_UHML_hIBS_cov.manhattan.png blup_UHML_hIBS_cov

cat UHML_EMMAX_hBN_cov.ps|awk -F'[\t_]' '{print $1"_"$2,$1,$2,$4}' OFS='\t'|sed '1iSNP\tCHR\tBP\tP' > blup_UHML_hBN_cov.manhattan.txt &

Rscript qqman.script2.R blup_UHML_hBN_cov.manhattan.txt blup_UHML_hBN_cov.manhattan.png blup_UHML_hBN_cov

Rscript qqman.script2.R blup_UHML_hIBS.manhattan.txt blup_UHML_hIBS.manhattan.png blup_UHML_hIBS

```





# 2021-12-03 Friday

附表symbol discription，通过metascape添加。


马峙英老师的数据字母开头的样本有419个。

```
(mashr) 11:56 zyqi@login02:~/cotton_fiber/gwas/trait_blup_376_filter_ranLineRep
$cat BLUP_UHML_out|awk '{print $1,$5,$10,$11,$12}' OFS='\t'|les
```
数量性状似乎不看OddsRatio


未解决的问题：
obsidian代码注释快捷键
mac 连接ftp


zotero wikipedia有时候不会保存snapshot #zotero





# 2021-12-04 Saturday

=SUBSTITUTE(L2,"_new","")

```shell
cat srr_select.txt|while read -a i;do mv srr_sample/${i[0]}.sra srr_sample/${i[1]}.sra;done
```

/data/cotton/zyqi/ma_2021_ng





# 2021-12-05 Sunday



```bash
for i in `ls SRR_5*/SRR*/*sra`;do chmod 775 $i;done
# fastq-dump --gzip --split-3 /data/cotton/zyqi/ma_2021_ng/srr_sample/$i.sra -O /data/cotton/zyqi/ma_2021_ng/clean_data
```


看看文件删除的情况
周洁的账号 snp_calling_sampleName_00_aa.sh

我自己的
sh snp_calling_sampleName_00_ab.sh




2021年12月06日
张宇琪：snp_calling_sampleName_01.sh
张宇琪：snp_calling_sampleName_00_ac.sh

周洁：snp_calling_sampleName_03.sh
我自己：snp_calling_sampleName_02.sh


2021-12-05 20:01:34
mdnotes导出的文件，我要使用cmd+shift+v复制到obsidian里，才不会有三个空行。
#zotero #obsidian





# 2021-12-06 Monday

 

BWA-MEM (Li, 2013; Li and Durbin, http://bio-bwa.sourceforge.net/ 2009)







# 2021-12-07 Tuesday

重命名之后，我的备注消失了。原来的文件名是Full Text PDF。我也不知道为啥。也可能是预览的bug #zotero 

Rapid epistatic mixed-model association studies by controlling multiple polygenic effects 可能可以用自然群体检测上位性





# 2021-12-10 Friday


因为Disk quota exceeded的问题需要重新下载sra数据


/data/cotton/zyqi/ma_2021_ng

找出那些样品下载成功了（但其实不能以report为准）
```shell
grep "无法删除" vcf_call*/*/*err > undeleated_sra.txt
cat undeleated_sra.txt|awk -F'"' '{print $2}' > deleated_path.txt
cat deleated_path.txt|while read i;do rm $i;done

l *html|awk '{print $9}'|sed 's/.html//' > ../succeed.txt
```

哪些样品没有成功
```shell
cat srr_select.txt|awk '{print $2,$1}' OFS='\t' > srr_select_1.txt
sed -i 's/^/\^&/' succeed.txt #sed 行首加字符
grep  -f succeed.txt srr_select_1.txt > succeed_srr.txt
grep  -v -f succeed.txt srr_select_1.txt > unsucceed_srr.txt
```


## 重新下载数据

```shell
mkdir d_6;cd d_6;bsub -q smp -o %J.out -e %J.err -m s006 "sh ../download_prefetch_05";cd ..

mkdir d_2;cd d_2;bsub -q smp -o %J.out -e %J.err -m s003 "sh ../download_prefetch_01";cd ..

mkdir d_3;cd d_3;bsub -q smp -o %J.out -e %J.err -m s001 "sh ../download_prefetch_02";cd ..

mkdir d_4;cd d_4;bsub -q smp -o %J.out -e %J.err -m s003 "sh ../download_prefetch_03";cd ..

mkdir d_5;cd d_5;bsub -q smp -o %J.out -e %J.err -m s001 "sh ../download_prefetch_04";cd ..

mkdir d_1;cd d_1;bsub -q smp -o %J.out -e %J.err -m s001 "sh ../download_prefetch_00";cd ..
l d_1/*/*tmp|wc -l
```

/data/cotton/zyqi/ma_2021_ng_remain
都在下载了，642个
有report 439，/data/cotton/zyqi/ma_2021_ng/report



看大于1G的vcf.gz的个数，call成功的很少。原因是fasterq-dump这一步得到的fastq文件不完整，不完整的原因是Disk quota exceeded

```shell
#文件路径 
ls ./vcf_call*/*/*gz|xargs -I {} -P 5  find {}   -size +1G -type f |wc -l
```





# 202112-13 Monday

## call SNP
yqzhang:282，/public/home/yqzhang/zyqi/ma_2021_ng_remain
周洁：360个在call SNP，/public/home/jzhou/zyqi/ma_2021_ng_remain/data



## sra剩余数据下载并call SNP

安装sratoolkit 2.11.0
wget https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.11.0/sratoolkit.2.11.0-centos_linux64.tar.gz

使用prefetch.2.11.0进行下载（注意版本号）

### 张宇琪
下载路径：
/public/home/yqzhang/zyqi/download/ma_2021_ng_remain_1/d_1，投递
bsub -q smp "sh download.sh"
/public/home/yqzhang/zyqi/download/ma_2021_ng_remain_1/d_0

下载完成，call SNP，共160个样本
/public/home/yqzhang/zyqi/download/ma_2021_ng_remain_1/
```bash
mkdir data clean_data report SNP_call
mv d_*/SRR*/*sra data
cd data
l *sra|awk '{print $9}'|sed 's/.sra//' > sample_name.txt
sh snp_calling_sampleName_02.sh
```



### 卢思凡
下载路径：/public/home/sflu/zyqi/download/ma_2021_ng_remain_1

```bash
#检查文件大小是不是都超过了4G
ls ./d_*/SRR*/*sra|xargs -I {} -P 5  find {}   -size +4G -type f |wc -l
```


下载完成，call snp，共279个样本
路径：/public/home/sflu/zyqi/download/ma_2021_ng_remain_1

```bash
mkdir data clean_data report SNP_call
mv d_*/SRR*/*sra data
cd data
l *sra|awk '{print $9}'|sed 's/.sra//' > sample_name.txt
# 分两个队列投递
sh snp_calling_sampleName_02.sh
sh snp_calling_sampleName_02_normal.sh
```


## 注意事项
module load sratoolkit/2.9.6
会统一下载在~/ncbi/public/sra路径
集群上的prefetch可能做过什么配置，或者版本过低，我多个sra一起下，[会报错](https://app.yinxiang.com/shard/s65/nl/17152587/c8f37734-dab2-4d66-8687-1587ed6b75bd/)



## 清理目录
/data/cotton/zyqi
/data/cotton/zyqi/archives，fusion文件需要整理
8.0T	fusion





# 2021-12-14 Tuesday


## 日常任务
原始测序数据上传，投递在normal节点
~/ga/raw_data


unrar解压，要加x
./rar/unrar x gmat-1.01.rar



## 上位性软件gmat安装

```
(new) 23:07 zyqi@mn02:/data/cotton/zyqi
$gmatrix
gmatrix: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by gmatrix)
gmatrix: /lib64/libstdc++.so.6: version `CXXABI_1.3.8' not found (required by gmatrix)
gmatrix: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by gmatrix)
```



strings /usr/lib64/libstdc++.so.6 | grep CXXABI
报错的原因是libstdc++.so.6 版本过低，
https://www.cnblogs.com/SummerinShire/p/11840810.html
https://blog.csdn.net/troytom/article/details/81557408



```shell
# 创建一个新的conda环境
conda create --name gmat
conda activate gmat


conda install libgcc
# 找出最新版动态库的路径
find . -name "libstdc++.so*"

#查看是否包含了CXXABI_1.3.8
strings $HOME/anaconda3/pkgs/libstdcxx-ng-11.2.0-he4da1e4_8/lib/libstdc++.so.6 | grep CXXABI

# 使用gmatrix，export动态库路径
export LD_LIBRARY_PATH=$HOME/anaconda3/pkgs/libstdcxx-ng-11.2.0-he4da1e4_8/lib

```


## 分成100个子任务投递
2021年12月25日

### 张宇琪

传数据
```shell
scp -P 22333 pheno_fs.txt   yqzhang@211.69.141.142:/public/home/yqzhang/zyqi/epistasis/FS
```

批量生成脚本
```shell
for i in {1..100};do echo "remma --epis abya --bfile /data/cotton/zyqi/epistasis/filter2_Q1000_SNPs_joint_376_MAF0.05_26_number --data pheno_enlongation.txt --grm cotton.agrm.id_fmt,cotton.aagrm.id_fmt --covar COV1,COV2,COV3 --trait trait --continue --maxiter 0 --out enlongation --parallel $i,100" > batch_run/run_$i.sh;done
```

投递
```shell
export LD_LIBRARY_PATH=/public/home/yqzhang/software/anaconda3/pkgs/libstdcxx-ng-9.1.0-hdf63c60_0/lib/
for i in {1..100};do bsub -n 1 -q q2680v2 -R "rusage[mem=30GB]" -J $i -o ${i}_%J.out -e ${i}_%J.err "sh batch_run/run_$i.sh";done
```



## UI
两个账号下都投了

卢思凡，normal队列
/public/home/sflu/zyqi/epistasis


high 队列，总共用了三百核了，看看能不能跑上
/public/home/yqzhang/zyqi/epistasis/UI


## FL
周洁的账号上也投了
/public/home/jzhou/zyqi/epistasis/FL





# 2021-12-15 Wednesday

gmatrix模块构建的加性基因组关系矩阵

```bash
export LD_LIBRARY_PATH=$HOME/anaconda3/pkgs/libstdcxx-ng-11.2.0-he4da1e4_8/lib
gmatrix --bfile data/filter2_Q1000_SNPs_joint_376_MAF0.05_26_number --grm agrm --out cotton
```

任务投递
for i in {00..09};do bss "sh calc_2_$i";done




测试parallel结果会不会不同
~/cotton_fiber/gmat_test
~/cotton_fiber/gmat_test_parallel
确有不同，提了issue


其他性状也在跑了，没有用--parallel





# 2021-12-16 Thursday

全基因组选择育种知识梳理
育种值的定义：



~/cotton_fiber/CNN
git clone https://bitbucket.org/bucklerlab/maize_yield_pred.git
wget https://data.cyverse.org/dav-anon/iplant/home/washjake/CNN_data_for_ms11May2021.tar.gz
tar xvzf CNN_data_for_ms11May2021.tar.gz

~/cotton_fiber/CNN/data/Genotype_data，只用了2万多个标记





# 2021-12-19 Sunday


## 马峙英SNP calling

没call出来的样本

```bash
ls SNP_call/*/*gz|xargs -I {} -P 5  find {}   -size +3G -type f |wc -l
```

2021-12-16T06:27:47 fasterq-dump.2.9.6 fatal: SIGNAL - Segmentation fault

周洁
/public/home/jzhou/zyqi/ma_2021_ng_remain/SNP_call
SRR14552747

张宇琪
两个样本有问题，/public/home/yqzhang/zyqi/download/ma_2021_ng_remain_1
SRR14552575
SRR14552577
一个样本有问题，/public/home/yqzhang/zyqi/ma_2021_ng_remain/SNP_call
SRR14470576

卢思凡
/public/home/sflu/zyqi/download/ma_2021_ng_remain_1/SNP_call
SRR14470404
SRR14470050
SRR14470422
SRR14470015

```bash
ls SNP_call/*/*gz|xargs -I {} -P 5  find {}   -size +1G -type f |xargs -I {} cp -p {} /data/cotton/zyqi/ma_2021_ng_vcf
ls SNP_call/*/*gz.tbi|xargs -I {} -P 5  find {} -type f |xargs -I {} cp -p {} /data/cotton/zyqi/ma_2021_ng_vcf
```

bsub -q q2680v2 -o %J.out -e %J.err "sh cp.sh"



剩余8个数据下载

/data/cotton/zyqi/ma_2021_remain_8

call SNP
/data/cotton/zyqi/ma_2021_remain_8/SNP_call





# 202112-20 Monday

~/ga/raw_data_RNAseq

```bash
bsub -q high -o 4DPA_%J.out -e 4DPA_%J.err "sh 4DPA_upload.sh"
bsub -q high -o 8DPA_%J.out -e 8DPA_%J.err "sh 8DPA_upload.sh"
bsub -q high -o 12DPA_%J.out -e 12DPA_%J.err "sh 12DPA_upload.sh"
bsub -q high -o 16DPA_%J.out -e 16DPA_%J.err "sh 16DPA_upload.sh"
bsub -q high -o 20DPA_%J.out -e 20DPA_%J.err "sh 20DPA_upload.sh"
```





# 2021-12-21 Tuesday

创建一个新的环境，GP基因组预测 #conda
conda create -n GP r-base=3.6 r-irkernel r-tidyverse jupyterlab r-languageserver python=3.7

jupyter kernelspec list

## 添加内核
添加R内核到jupyterlab
IRkernel::installspec(name='ir_gtex',displayname='R (gtex)')

添加python内核到jupyterlab
python -m ipykernel install --user --name python3_mashr --display-name "Python3 (mashr)"



kernelspec中文件下的kernel是name决定的，如（ir33），展示在jupyter上的kernel名字是displayname 决定的，如（R 3.3）

## 卸载内核
[印象笔记jupyter内核问题](https://app.yinxiang.com/shard/s65/nl/17152587/c6e21058-5fb3-4873-8304-d57e9728811d/)
jupyter kernelspec remove ir_gtex
jupyter kernelspec remove python3_gtex

conda activate GP





# 2021-12-22 Wednesday

卸载的命令
conda uninstall r-base









我想问他怎么使用这个模型

Thank your for you reply. I have another question. In your example data, I noticed that I should put the individulas 
prediction equations





# 2021-12-29 Wednesday

## 上位性，过滤
/public/home/yqzhang/zyqi/epistasis/FS

```shell
l *[0-9].res|awk '{print $9}' > output_file.txt
mkdir sig
cat output_file.txt|while read i;do bsub -q q2680v2 -o filter_%J.out -e filter_%J.err "cat $i|awk '\$19<1E-10' > sig/$i.sig";done
```

没跑完的单独过滤了
cat fs.abya.295621_310626.res|awk '$19<1E-10' > sig/fs.abya.295621_310626.res.sig

/public/home/jzhou/zyqi/epistasis/FL
cat uhml.abya.136448_150500.res|awk '$19<1E-10' > sig/fuhml.abya.136448_150500.res.sig





# 2021-12-30 Thursday

#FUSION
/data/cotton/zyqi/archives/fusion/blup_376/0D
/data/cotton/zyqi/gwas_419/twas

二倍体fusion路径
/data/cotton/zyqi/Ga/Ga_fusion/fusion_new_1


构建输入文件
~/cotton_fiber/gwas/trait_blup_376_filter_ranLineRep

/data/cotton/zyqi/Ga/Ga_GWAS_new/phe1/zscore


#awk  处理基因注释文件
```shell
cat A2_anno.txt|awk -F'\t' '{split($1,a,/\./);$1=a[1];split($2,b,/\./);$2=b[1];print}' OFS='\t'|sort |uniq > A2_anno_gene.txt 
```

 

GWAS was performed separately for each cohort by GCTA v.7.93.2 (ref. 77) with the mixed linear model. #GCTA 是做GWAS的软件 [[@wei2021]]





# 2021-12-31 Friday

#find 某一时间点之前的所有文件，然后移动
~/cotton_fiber/gwas
```shell
find . -maxdepth 1  -mtime +200 -exec mv {} 349_result_collect \;
```
