# 2021-09-01 Wednesday


![[二倍体分析#任务]]



conda install -c bioconda bioconductor-genomicfeatures
conda install -c bioconda bioconductor-chipseeker

conda install -c bioconda bioinfokit # 用于转gff3 转 gtf
https://www.reneshbedre.com/blog/gffgtf.html
https://cloud.tencent.com/developer/article/1625209


不知道Distal intergenic 怎么定义的
 - TSS前面的，和TTS后面的，都是基因间区。TSS前面有一段被拿来算在promoter里面了，TTS后面一段归入immediate downstream了。前面这两块都是近邻，那么剩下的就是distal intergenic了。https://mp.weixin.qq.com/s/MqpfgkMJSFj0pYwcEjV9kQ

#tracking 
09:39 一份讲课稿件：

- It has already taught us a few lessons, and the one I have chosen to stress in this talk is the following. We shall do a much better programming job, provided that we approach the task with a full appreciation of its tremendous difficulty, provided that we stick to modest and elegant programming languages, provided that we respect the intrinsic limitations of the human mind and approach the task as Very Humble Programmers.
- 10:21 印象笔记似乎成了一个记录bug的笔记。PPT上及时记录分析路径。
- 11:24 eoffice导出的字体：_Helvetica_（赫尔维提卡体）
- [x] 11:28 算纤维长度
- [x] 11:36 找时间还得做一些渐渗的染色体分布图
- ppt导出PDF，每次生成的文件md5值不一样。缩放之后的PDF有白边
- ppt压缩图片，原始质量比高保真文件更大
- 纤维长度，有些人为的因素。J98 20D 0.26cm
- [x] eQTL注释，https://mp.weixin.qq.com/s/vWTf59KDs1lp_sQXjEhI_g
- [ ] 是否富集在启动子区
- [ ] ChIPseeker这结果对象有哪些信息可以探索一下





# 2021-09-02 Thursday



#tracking 
- 09:06 ggplot的默认字体是Helvetica？可能依赖操作系统。
- [ ] 14:23 ggplot如何设置输出字体，Arial
-  图形（+代码）保存到印象笔记
- [ ]  怎么把R换成英文环境 [[2021-09-09]]

Reversing the order of a ggplot2 legend
https://aosmith.rbind.io/2018/01/19/reversing-the-order-of-a-ggplot2-legend/
`plotAnnoBar(peakAnnoList) + scale_fill_manual(values=cl1[10:1],guide = guide_legend(reverse = TRUE)) `



conda install -c bioconda bioconductor-biovizbase
画染色体图

conda install -c bioconda bioconductor-ggbio


txdb = makeTxDbFromGFF('change.Chr_genome_final_gene.gff3')


Make a GRanges object from a data.frame or DataFrame





# 2021-09-03 Friday



/data/cotton/zyqi/vcf_call/result_100_0.5.txt



![[Pasted image 20210903102508.png|300]]

![[Pasted image 20210904085843.png|300]]

渐渗的图画不好看

```R
scale_fill_gradient(low = "white", high = "red", na.value = NA) +
scale_color_gradient(low = "white", high = "red", na.value = NA)
    
```




热图颜色，连续

```R
mycolor <- c( "#FFFFFFFF","#FDE725FF","#9FDA3AFF","#4AC16DFF","#1FA187FF","#277F8EFF","#365C8DFF","#46337EFF","#440154FF")
```





## hotscan软件安装

conda install perl

sudo cpan


File::chdir


Math::GSL::SF


export all_proxy=socks5://127.0.0.1:1080


conda install perl=5.26.2

降级是不是一种方法


本地安装Math::GSL::SF失败。

 

Math::GSL::SF  
AnyEvent::ForkManager
File::Path

检查模块是否安装成功
perldoc Math::GSL::SF

conda create -n hotspot
conda install perl=5.26.2


```bash
bsub -q normal -J  hotspot -n 5 -R span[hosts=1] -e test.err "
module load GSL
perl /data/cotton/zhenpingliu/LZP_fiberFullPopulationRNAseq/software/hot_scan-master/hot_scan.pl  -m 500 -c ../chromsome_len_human.txt  -e All_eVarianvce_event.txt -p 5 -s 0.05 -o hotspot_out"
```

cpan -l

16:14
我在conda new环境下使用cpan -l的时候，创建了~/perl5目录



最后让我在~/.bashrc里添加环境变量
```bash
# local::lib is installed. You must now add the following environment variables
# to your shell configuration files (or registry, if you are on Windows) and
# then restart your command line shell and CPAN before installing modules:


PATH="/public/home/zyqi/perl5/bin${PATH:+:${PATH}}"; export PATH;
PERL5LIB="/public/home/zyqi/perl5/lib/perl5${PERL5LIB:+:${PERL5LIB}}"; export PERL5LIB;
PERL_LOCAL_LIB_ROOT="/public/home/zyqi/perl5${PERL_LOCAL_LIB_ROOT:+:${PERL_LOCAL_LIB_ROOT}}"; export PERL_LOCAL_LIB_ROOT;
PERL_MB_OPT="--install_base \"/public/home/zyqi/perl5\""; export PERL_MB_OPT;
PERL_MM_OPT="INSTALL_BASE=/public/home/zyqi/perl5"; export PERL_MM_OPT;

# Would you like me to append that to /public/home/zyqi/.bashrc now? [yes]
```




```bash
cpanm Math::GSL::SF
cpanm AnyEvent::ForkManager File::Path
```


```
$cpanm Math::GSL::SF
!
! Can't write to /usr/local/share/perl5 and /usr/local/bin: Installing modules to /public/home/zyqi/perl5
! To turn off this warning, you have to do one of the following:
!   - run me as a root or with --sudo option (to install to /usr/local/share/perl5 and /usr/local/bin)
!   - Configure local::lib your existing local::lib in this shell to set PERL_MM_OPT etc.
!   - Install local::lib by running the following commands
!
!         cpanm --local-lib=~/perl5 local::lib && eval $(perl -I ~/perl5/lib/perl5/ -Mlocal::lib)
!
--> Working on Math::GSL::SF
```

cpanm Math::GSL::SF需要先`module load GSL`





```shell
sed -i '1s/^/Gene\t&/g' merge_df.txt
```

 #sed

https://www.cnblogs.com/aaronwxb/archive/2011/08/19/2145364.html





# 2021-09-04 Saturday



my_bash_history 做了备份，存放在~/bash_history_back  #集群



eQTL, ~/cotton_fiber/data/nominal

2021年09月09日：
- [x] add ID, real TSS distance #二倍体项目

二倍体：
/data/cotton/zyqi/Ga_data/nominal





```
(new) 21:39 zyqi@login02:~/cotton_fiber/locus_plot
$cat README.txt
使用gtex环境

1. gff2转成gtf用的是python包。bioinfokit。
>>> from bioinfokit.analys import gff
>>> gff.gff_to_gtf(file="Athaliana_167_TAIR10.gene_chr1.gff3")

2. 用的当前目录下的annotation脚本

3. 标准化后的表达量去掉了不在染色体上的基因
```



gwas lead snp
~/cotton_fiber/gwas/trait_blup_376_filter_ranLineRep






平均每个样本的渐渗长度为 5.01M
analysis_0714.ipynb


TWAS    表达量FPKM做log转换

# 2021-09-05 Sunday



## hot spot



hotspot第5列为这个区域eQTL的数目。用的什么检验说这是一个hotspot？

~/work/ga/hotspot_all/tracks_c 



#awk #正则
cat 12D_all.bed.500.bc.un.track.bed|awk '$5!~/[0-9]/'
正则匹配要加斜杠

[[awk]]





# 2021-09-06 Monday

- [ ] circos 调控热点区域，links，ggbio


![[Pasted image 20210906163212.png|300]]


应该在编写脚本之前就写好了测试。这该如何操作？
https://mp.weixin.qq.com/s/tbw2pc51Koir6A5Be2PYfw


awk默认按照字符串比较的，如果要比较数值，需要先强制类型转换，即 $1+0，转换成整数。

```sh
awk 'BEGIN {max = 0} {if ($1+0 > max+0) max=$1} END {print "Max=", max}' test.txt  
Max= 118


awk 'BEGIN {min = 1000} {if ($1+0 < min+0) min=$1} END {print "Min=", min}' test.txt  
Min= 9


cat test.txt|awk '{sum+=$1} END {print "Sum= ", sum}'  
Sum=  236

cat test.txt|awk '{sum+=$1} END {print "Avg= ", sum/NR}'  
Avg=  39.3333
```



chr11 21789156 21789157
落在什么位置？

- [ ] 变异注释
~/ga/annovar




![[Pasted image 20210906225319.png|300]]


~~conda install -c bioconda bioconductor-regioner~~

Grange
2:166746367





# 2021-09-09 Thursday

mashr: Multivariate Adaptive Shrinkage
https://cran.r-project.org/web/packages/mashr/index.html

https://cran.r-project.org/web/packages/mashr/vignettes/intro_mash.html

```
print(sessionInfo())
# R version 3.6.2 (2019-12-12)
# Platform: x86_64-apple-darwin15.6.0 (64-bit)
# Running under: macOS Catalina 10.15.7
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
# LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] mashr_0.2.50 ashr_2.2-51 
# 
# loaded via a namespace (and not attached):
#  [1] Rcpp_1.0.5        knitr_1.26        magrittr_1.5      SQUAREM_2017.10-1
#  [5] lattice_0.20-38   rlang_0.4.5       stringr_1.4.0     plyr_1.8.5       
#  [9] tools_3.6.2       grid_3.6.2        xfun_0.11         rmeta_3.0        
# [13] irlba_2.3.3       invgamma_1.1      htmltools_0.4.0   yaml_2.2.0       
# [17] abind_1.4-5       digest_0.6.23     assertthat_0.2.1  softImpute_1.4   
# [21] mixsqp_0.3-46     Matrix_1.2-18     evaluate_0.14     rmarkdown_2.3    
# [25] stringi_1.4.3     compiler_3.6.2    truncnorm_1.0-8   mvtnorm_1.0-11
```



[[2021-09-02]]
- [x] R改语言，locale？ #R


```
> print(sessionInfo())
R version 4.0.5 (2021-03-31)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS/LAPACK: /Users/zhengyangqi/opt/miniconda3/envs/r1/lib/libopenblasp-r0.3.17.dylib

locale:
[1] zh_CN.UTF-8/zh_CN.UTF-8/zh_CN.UTF-8/C/zh_CN.UTF-8/zh_CN.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

loaded via a namespace (and not attached):
[1] compiler_4.0.5
```

R英文环境

https://cran.r-project.org/bin/macosx/RMacOSX-FAQ.html#Internationalization-of-the-R_002eapp

R解释器：`system("defaults write org.R-project.R force.LANG en_US.UTF-8")`。没有用，可能改的不是conda R


如果只在~/.Renviron添加`LANGUAGE=en` locale 不会改变。添加`LANG=en_US.UTF-8`才可以

https://qastack.cn/programming/13575180/how-to-change-language-settings-in-r
终端：export LANG=en_US.UTF-8 也可以

## R配置
https://bookdown.org/yihui/r-ninja/setup.html
我们主要介绍两个配置文件，`.Renviron`和`.Rprofile`






二倍体[[2021-09-04|加ID]] #二倍体项目





# 2021-09-10 Friday

 

14.1 Estimating cross tissue activity of QTLs





# 2021-09-11 Saturday

jupyter lab 的撤销不好用，不知道撤销到哪一步了，没有跳转 



chr11 21789157 position
```
(new) 11:19 zyqi@login02:~/ga/annovar
$grep 21789157 filter2_Q600_SNPs_joint_216.variant_function
intergenic	Garb_11G018120(dist=14155),Garb_11G018130(dist=14822)	Chr11	21789157	21789157	A	G	0.08621	3331.49	6
```



安装pyqtl
https://github.com/broadinstitute/pyqtl



## mashr
https://cran.r-project.org/web/packages/mashr/vignettes/eQTL_outline.html
或
https://stephenslab.github.io/mashr/articles/eQTL_outline.html


软件安装 new
install.packages("mashr")



eQTL summary statistics formatting
This workflow converts fastqtl eQTL analysis summary statistics text output to formats more friendly to R analysis. In particular:

It converts single study results to HDF5 format grouped by genes.
It combines multiple studies into one single HDF5 file. In the context of GTEx each study is result from one tissue.
For MASH analysis in particular, it extracts from the complete data a subset of results to compute data driven MASH prior covariance, and to fit the MASH mixture model.





# 2021-09-12 Sunday

## jupyter lab添加不同环境下的内核 
#jupyterlab 
conda activate gtex

conda install r-base=4.0.5 r-irkernel r-tidyverse r-languageserver

jupyter kernelspec list

https://blog.csdn.net/weixin_46021869/article/details/115264181
IRkernel::installspec(name='ir_gtex',displayname='R (gtex)')

```
> IRkernel::installspec(name='ir_gtex',displayname='R (gtex)')
[InstallKernelSpec] Installed kernelspec ir_gtex in /public/home/zyqi/.local/share/jupyter/kernels/ir_gtex
```

kernelspec中文件下的kernel是name决定的，如（ir33），展示在jupyter上的kernel名字是displayname 决定的，如（R 3.3）。


python -m ipykernel install --user --name python3_gtex --display-name "Python3 (gtex)"

```
(gtex) 11:31 zyqi@login02:~
$python -m ipykernel install --user --name python3_gtex --display-name "Python3 (gtex)"
Installed kernelspec python3_gtex in /public/home/zyqi/.local/share/jupyter/kernels/python3_gtex
```



## mashr
conda install sos -c conda-forge

https://github.com/stephenslab/gtexresults/blob/master/workflows/gtex6_mash_analysis.ipynb


pip install tables

if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("rhdf5")



conda install -c conda-forge hdf5

conda install networkx">=2.5"


wget https://raw.githubusercontent.com/stephenslab/gtexresults/master/workflows/fastqtl_to_mash.ipynb

workflows/fastqtl_to_mash.ipynb  失败
![[Pasted image 20210912175818.png]]



## 其他

更改python r 版本
conda create -n mashr r-base=3.6 r-irkernel r-tidyverse jupyterlab r-languageserver sos python=3.7

conda activate mashr

conda install -c conda-forge hdf5
conda install -c bioconda bioconductor-rhdf5
conda install scipy
conda install pandas

install.packages("mashr")


bsub -J mashr -q high  -o %J.out -e %J.err -n 36 "sh run1.sh"

成功


![[Pasted image 20210912193056.png]]

IRkernel::installspec(name='ir_mashr',displayname='R (mashr)') #jupyterlab 

```
> IRkernel::installspec(name='ir_mashr',displayname='R (mashr)')
[InstallKernelSpec] Installed kernelspec ir_mashr in /public/home/zyqi/.local/share/jupyter/kernels/ir_mashr
```




明日计划

lfsr作图





# 2021-09-13 Monday

## 跑tensorQTL

- [ ] 生成基因的TSS位置信息

~/cotton_fiber/genome/Garboreum_genome_HAU_v1.0/change.Chr_genome_final_gene_0based_TSS.bed

```shell
cat change.Chr_genome_final_gene_1based.bed |awk '{if ($6=="+") print $1,$2,$4;else print $1,$3,$4}' OFS='\t' > change.Chr_genome_final_gene_1based_TSS.bed
```


- [ ] 准备基因表达bed文件

~/ga/expression/expression_data


```shell
awk 'NR==FNR{a[$4]=$1"\t"$2"\t"$3;next}{$1=a[$1]"\t"$1;print}' ~/cotton_fiber/genome/Garboreum_genome_HAU_v1.0/change.Chr_genome_final_gene_0based_TSS.bed Ga_8DPA_AllSamples_201_FPKM_AllStages_norm.txt > Ga_8DPA_AllSamples_201_FPKM_AllStages_norm.bed
```

### 参考

表达bed文件格式要求

```
#Chr start end ID UNR1 UNR2 UNR3 UNR4
chr1 173863 173864 ENSG123 -0.50 0.82 -0.71 0.83  
chr1 685395 685396 ENSG456 -1.13 1.18 -0.03 0.11  
chr1 700304 700305 ENSG789 -1.18 1.32 -0.36 1.26
```


- Phenotypes must be provided in BED format, with a single header line starting with `#` and the first four columns corresponding to: `chr`, `start`, `end`, `phenotype_id`, with the remaining columns corresponding to samples (the identifiers must match those in the genotype input). The BED file should specify the center of the _cis_-window (usually the TSS), with `start == end-1`. A function for generating a BED template from a gene annotation in GTF format is available in [pyqtl](https://github.com/broadinstitute/pyqtl) (`io.gtf_to_tss_bed`).



## independent eQTL 鉴定
2021年09月14日


R包`qvalue`没装，导致没有q value

`pip install rpy2`

```R
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("qvalue")
```

跑'cis' mode
`KeyError: 'Contig00200'`，有的染色体变异文件中没有

把没有在染色体上的基因删掉。221个。
`zcat Ga_4DPA_AllSamples_202_FPKM_AllStages_norm.bed.gz|grep Contig|wc -l`



任务投递
```
bsub -q gpu -J 4D_GPU -o %J.out -e %J.err "sh indep_GPU.sh"
bsub -q gpu -J 8D_GPU -o %J.out -e %J.err "sh indep_GPU_8D.sh"
bsub -q gpu -J 12D_GPU -o %J.out -e %J.err "sh indep_GPU_12D.sh"
bsub -q gpu -J 16D_GPU -o %J.out -e %J.err "sh indep_GPU_16D.sh"
bsub -q gpu -J 20D_GPU -o %J.out -e %J.err "sh indep_GPU_20D.sh"
```


gpu01满了之后会投递到gpu02。gpu02不可用。 #集群 

24000个基因用GPU跑，大概需要3个小时。

![[Pasted image 20210914113102.png]]

8D 在gpu02节点上，就用不了GPU

```
JOBID      USER    STAT  QUEUE      FROM_HOST   EXEC_HOST   JOB_NAME   SUBMIT_TIME
42185051   zyqi    RUN   q2680v2    login02     sg21        jupyter    Sep 13 15:51
42228731   zyqi    RUN   gpu        login02     gpu01       4D_GPU     Sep 14 11:17
42228826   zyqi    RUN   gpu        login02     gpu02       8D_GPU     Sep 14 11:28
42228931   zyqi    RUN   q2680v2    login02     sg22        *PU_12D.sh Sep 14 11:35
42228942   zyqi    RUN   high       login02     c01n14      16D        Sep 14 11:36
42228953   zyqi    RUN   high       login02     c01n05      20D        Sep 14 11:37
```

23:07
- 12D, 16D GPU只有2个小时不到，~/ga/tensorQTL/run_tensorQTL/GPU
- 16D high 7个小时。12D q2680v2 12个小时都没跑完



## 软件安装
08:38 整理了一下印象笔记，重新命名和移动了一些文件夹

https://github.com/broadinstitute/pyqtl  Python包 qtl 一直在更新

mashr环境安装包
- install.packages("pheatmap")

- pip install bioinfokit # gff转gtf，画图

- python -m ipykernel install --user --name python3_mashr --display-name "Python3 (mashr)" # 将mashr环境下的内核加到jupyter lab中




pip uninstall qtl 不会卸载对应的依赖

plink 转格式 [[plink]]

- [ ] vcf文件添加ID

```sh
zcat filter2_Q600_SNPs_joint_216.vcf.gz|awk '{if(/^#/){print}else{$3=$1"_"$2;print}}' OFS='\t' > filter2_Q600_SNPs_joint_216_addID.vcf
```


module load HTSlib/1.9
bgzip filter2_Q600_SNPs_joint_216_addID.vcf
tabix filter2_Q600_SNPs_joint_216_addID.vcf.gz

- [ ] 用plink转vcf文件

~/ga/tensorQTL/Genotypes

module load plink
plink --vcf filter2_Q600_SNPs_joint_216_addID.vcf.gz --out filter2_Q600_SNPs_joint_216_addID --allow-no-sex --allow-extra-chr --make-bed





# 2021-09-14 Tuesday

## mashr
- [x] 拿最显著的eQTL跑的mashr。 #mashr

会报错。
~/ga/tensorQTL/permute/fastqtl_to_mash_output/


## tensorQTL
通过tensorQTL得到top cis-QTL和independent cis-QTL。top cis-QTL做效应


用tensorQTL做nominal生成的是.parquet文件，可用python打开。
https://github.com/broadinstitute/tensorqtl

~/ga/tensorQTL/run_tensorQTL/nominal

```python
import pandas as pd
df = pd.read_parquet('12DPA.cis_qtl_pairs.Chr07.parquet')
```



## 文献阅读
These empirical P values were subsequently corrected for multiple testing across genes using Storey’s q value method. To identify the list of all significant variant–gene pairs associated with eGenes, variants with a nominal _P_ value below the gene-level threshold were considered significant and included in the final list of variant–gene pairs. [[@aguet2017]]

q value < 0.05 作为显著的variant-gene pairs.



Comparison of eQTL effect sizes
In order to compare effect sizes and cell type specificities of eQTLs, we utilized mash (Urbut et al., 2019) for top eVariants. The analysis was performed with eQTL beta coefficients and their standard errors as inputs. Randomly selected SNP-gene pairs (220,000) were used to fit the mash model. If the gene was not expressed in the subset, the beta coefficient was set to 0. Effect size estimates and LFSR outputted by mash were used as metrics of QTL magnitude and significance. [[@ota2021]]



 

14.1 Estimating cross tissue activity of QTLs [[@thegtexconsortium2020]]  #mashr 
In order to analyze the tissue specificity of QTLs, we used MashR [51] for every top cis-QTL per gene per tissue across all tissues where expression quantification was available. R v3.4.1 was used with mashr v0.2-6 and ashr v2.2-7. MashR was run using Z-scores as input, and 250,000 randomly selected SNP-gene pairs that were tested across all tissues were used to fit the mash model. Missing Z-score values (gene or splice quantification absent) were set to 0 and standard error to 1e6. Effect size estimates and local false sign rate (LFSR) generated by MashR were used as metrics of QTL magnitude and activity respectively. A LFSR < 0.05 was used as a threshold for significant QTL activity unless noted otherwise. 





# 2021-09-15 Wednesday

## plot using mashr paper scripts
~/work/mashr_paper/stephenslab-gtexresults-961b969

- 他的脚本用的Rmd格式，通过[workflowr](https://github.com/jdblischak/workflowr)这个包可以生成精美的html网页。可以搭配git使用。
网页路径：
`/Users/zhengyangqi/work/mashr_paper/stephenslab-gtexresults-961b969/docs/Tspecific.html`

- Tracking code development and connecting the code version to the results is critical for reproducibility. To start using Git, open the Terminal and type `git init` in your project directory.




### Tspecific.Rmd
We define "tissue-specific" to mean that the effect is at least 2-fold larger in one tissue than in any other tissue.

`pm.mash.beta <- pm.mash * standard.error` pm.mash相当标准化之后的效应？



`pm.mash.beta.norm <- het.norm(effectsize = pm.mash.beta)`
norm的过程: `x/x[which.max(abs(x))]`

```R
het.norm = function(effectsize) {
t(apply(effectsize,1,function(x){
x/x[which.max(abs(x))]
}))
}
```
![[Pasted image 20210915110754.png|400]]

### get_pairwise_sharing函数
`pairwise_sharing <- get_pairwise_sharing(m2)`

```R
function (m, factor = 0.5, lfsr_thresh = 0.05, FUN = identity) 
{
    R = get_ncond(m)
    lfsr = get_lfsr(m)
    S = matrix(NA, nrow = R, ncol = R)
    for (i in 1:R) {
        for (j in i:R) {
            sig_i = get_significant_results(m, thresh = lfsr_thresh, 
                conditions = i)
            sig_j = get_significant_results(m, thresh = lfsr_thresh, 
                conditions = j)
            a = union(sig_i, sig_j)
            ratio = FUN(get_pm(m)[a, i])/FUN(get_pm(m)[a, j])
            S[i, j] = mean(ratio > factor & ratio < (1/factor))
        }
    }
    S[lower.tri(S, diag = FALSE)] = t(S)[lower.tri(S, diag = FALSE)]
    colnames(S) = row.names(S) = colnames(m$result$PosteriorMean)
    return(S)
}
```


### 表达量分析
/Users/zhengyangqi/work/mashr_paper/stephenslab-gtexresults-961b969/analysis/ExpressionAnalysis.Rmd

In each case, the distribution functions are similar. This tells us that tissue-specific eQTLs are not simply reflecting tissue-specific
expression.

## mashr paper
代码资源
- [[@urbut2019#UrLs]]
- 这个URL https://doi.org/10.5281/zenodo.1296399 下可以看到有gtexresults-v2.0.zip 
- https://github.com/stephenslab/gtexresults github链接是自动更新的

Our analysis identifies more eQTLs than existing approaches, consistent with improved power. We show that although genetic effects on expression are extensively shared among tissues, effect sizes can still vary greatly among tissues. Some shared eQTLs show stronger effects in subsets of biologically related tissues (for example, brain-related tissues), or in only one tissue (for example, testis).





Distribution of eQTL sharing across tissues

file:///Users/zhengyangqi/work/mashr_paper/stephenslab-gtexresults-961b969/docs/SharingHist.html




## 其他
- Rstudio使用的R
  ```R
  > .libPaths()
  [1] "/Users/zhengyangqi/opt/miniconda3/envs/r1/lib/R/library"
  ```


- zotero和obsidian结果使用的问题：
~/Documents/mybook/Aim.json 没有自动更新 #computerStuff 

- VSCode 装了Docker v1.16.1，见该插件说明
The Docker extension makes it easy to build, manage, and deploy containerized applications from Visual Studio Code. It also provides one-click debugging of Node.js, Python, and .NET Core inside a container.

- 如何插入模板 #obsidian 
https://sspai.com/post/63481

- [ ] RStudio有没有类似的特性：悬浮窗口显示帮助函数页面





# 2021-09-16 Thursday

- [ ] 时期特异的eQTL检查，4D有点过多
- [ ] 基因功能区的富集分析
- [x] 拿到GWAS的原始结果，跑共定位



~/ga/annovar/filter2_Q600_SNPs_joint_216.variant_function





# 2021-09-17 Friday

## mashr
nominal表头不需要删除
cd ~/ga/mashr_tensorQTL

- [x] 重新跑
bsub -q smp -o %J.out -e %J.err -n 20 "sh run1.sh"



## 文献阅读

### LD score regression analysis
We initially fine-mapped our eQTLs with CAVIAR (Hormozdiari et al., 2014), made maximal posterior inclusion probability annotations (MaxCPP) as previously reported (Hormozdiari et al., 2018) and performed tissue-by-tissue stratified LD score regression (S-LDSC) (Bulik-Sullivan et al., 2015; Finucane et al., 2015) adjusting for functional annotation (“baseline model” provided by the developers).

To evaluate the effect of cell type-shared eQTLs, we made meta-analyzed annotation of 6 representative immune cell subset eQTLs (Mem CD4, Mem CD8, naive B, CL Mono, NK and Neu) using ==linear mixed model based== meta-analysis (Sul et al., 2013). We utilized fixed-effect P values after meta-analysis and fine-mapped eQTLs with CAVIAR. Subsequently, we obtained MaxCPP of meta-analysis annotation, which we call “Meta tissue” annotation. Meta tissue annotation showed significant heritability enrichment for some traits (Figure S4E). To consider tissue-shared eQTL effects together with tissue-specific eQTL effects, we jointly regressed LD scores for each cell eQTL annotation and meta-tissue annotation together with functional annotation.

Formatted GWAS summary statistics for LDSC by developers were downloaded from https://alkesgroup.broadinstitute.org/sumstats_formatted/.



### Fine-mapping of eQTLs
We fine-mapped eQTL variants of GTEx v8 whole blood data of European ancestry and ImmuNexUT classical monocytes and neutrophils. In order to assess loci whose eQTL causal variants were shared between 2 datasets, we limited our analysis to eVariants that were polymorphic in both, eGenes that were significant in both and the eGene whose top eVariant in GTEx had a P value < 1 × 10−5 in ImmuNexUT. These limitations resulted in 2,575 genes for classical monocytes and 2,359 genes in neutrophils. For fine-mapping of GTEx and our eQTLs, we ranked eVariants by P values in GTEx and analyzed the top 50 variants with CAVIAR (Hormozdiari et al., 2014) while permitting up to 5 causal variants. For joint fine-mapping analysis, we utilized eCAVIAR (Hormozdiari et al., 2016) for the same set of variants while permitting up to 5 causal variants. Our whole genome data were utilized for estimating LD for Japanese, and the 1000 genomes project phase 3 European samples were used for estimating LD for Europeans. We compared the number of variants in fine-mapped credible sets that were assumed to include all the causal variants with 95% confidence (95% credible set).

To assess the functional relevance of eQTLs, we utilized 18,734 autosomal variants that had significant impacts on regulatory activity using the survey of regulatory elements (SuRE) reporter technology (van Arensbergen et al., 2019) in K562 cells. We made 10,000 null sets of variants from 5,748,566 autosomal variants that were analyzed in SuRE study while adjusting for the distance to TSS with SuRE significant variants. The enrichment score was calculated by dividing the observed overlapped number of variants by the mean of the number of overlaps in the null sets. Confidence intervals were estimated using 10,000 times bootstrapping of randomly sampled null sets with replacement.


网页翻译助手





# 2021-09-18 Saturday

## 任务
mashr
~/ga/mashr_tensorQTL，完成

ecaviar
~/ga/expression/expression_data/ecaviar
bss "Rscript get_ecaviar_input.r 8DPA UHML"


## bug
跑 eCAVIAR 报错信息

```
$cat 42536873.err
Loading required package: Rcpp
Loading required package: RcppEigen
Error in `[.data.table`(x, r, vars, with = FALSE) :
  i is invalid type (matrix). Perhaps in future a 2 column matrix could return a list of elements of DT (in the spirit of A[B] in FAQ 2.14). Please report to data.table issue tracker if you'd like this, or add your comments to FR #657.
Calls: subset -> subset.data.table -> [ -> [.data.table
Execution halted
```

原因data.table的提子集的方式和data.frame不同
`eqtl_gene <- subset(eqtl,eqtl$V1==gene)` data.table的错误的写法


## 软件安装
~/software
`install.packages('RcppEigen')`，有warining，但能用 #computerStuff 
install.packages('plink2R-master/plink2R/',repos=NULL)


## 文件位置
tensorQTL的nominal结果移动位置
mv ~/ga/tensorQTL/nominal /data/cotton/zyqi
mv nominal tensorQTL_nominal







# 2021-09-20 Monday

![[Pasted image 20210920224421.png]]





有多少比例eVariant-gene位于loop的两端？


问下师弟，eQTL和基因表达的相关系数如何做？





# 2021-09-21 Tuesday

![[Pasted image 20210921090130.png]]


至少6月18号之前的历史记录我全没了 #集群



annot = qtl.annotation.Annotation('A2.gtf', verbose=False)
~/ga/locus_plot/addname.py



/Users/zhengyangqi/Documents/GitHub/gtex-v8/Fig7_celltype.ipynb

```python
import imp

imp.reload(qtl.map)

eqtl_df, cond1_df, cond2_df = qtl.map.get_conditional_pvalues(group_df, genotype_df, phenotype_df, covariates_df, phenotype_id=gene.id, window=1000000)
```


conda install  htslib

#jupyterlab 我是知道指定shift+tab可以看函数说明的?





# 2021-09-22 Wednesday

~/ga/expression/expression_data/ecaviar_0921/ecaviar_input/UHML/12DPA

~/cotton_fiber/locus_plot

phe1 UHML 没有显著的基因

~/ga/ecaviar/phe2
Rscript get_locus_plot_input.r"








看看结果是否一样。FS需不需要重跑？
结果是一样的。~/ga/ecaviar/phe2/ecaviar_result/UHML
![[Pasted image 20210922145316.png]]


LD 和相关系数的关系？绝对值


python locus_plot_argv.py 12DPA UHML Garb_05G038100 Chr05_85924943 200000

#python 路径不能用波浪线






~/ga/ecaviar/phe2
改 get_ecaviar_input.r



共定位用的LD是相关系数，locus plot用的LD是abs(r)

会不会是因为LD有负号导致计算的结果不怎么正确？


cur.genos = scale(genos$bed)
cur.LD = t(cur.genos) %*% cur.genos / (nrow(cur.genos)-1)


R计算相关系数，矩阵

~/ga/ecaviar/phe2_r2

19:57 换成R2
cur.LD = cur.LD * cur.LD








# 2021-09-23 Thursday

~/ga/ecaviar/locus_plot/wang_select_UHML

mashr和tensorQTL最大的不同是mashr 合并5个时期来鉴定eQTL，并计算效应。tensorQTL是每个时期单独计算，两个时期同个eGene的top evariant可能不同。

![[Pasted image 20210923082514.png]]

`cat change.Chr_genome_final_gene_1based.bed|sed 's/Chr[0]*//' > change.Chr_genome_final_gene_1based_number.bed` #sed


`sapply(str_split(test,'_'),function(x)paste(x[1],x[2],sep = '_'))` #R


相关性柱状图配色
`#2378b5 #d62a28 #f57f20 #a9a8a8 #d9d9d8
#d62a28`
“the standard deviation is zero”
Warning message in cor(x, y):

原因是表达量全为0



~/ga/expression/pca_plot_changeColor_2.r 使用log之后的表达量做PCA





# 2021-09-24 Friday

plink计算LD


基因型文件
~/uplandMAGIC/genotype_to_hmp


plink算LD
~/uplandMAGIC/filtering
~/uplandMAGIC/epistasis/plink_LD




~/uplandMAGIC/magic_new_210207/upland_magic/QTL_merge_210113/R2_merge


349份材料的可能在里面
~/cotton_fiber/ecaviar/to_trash_0711




module load BLAST+/2.9.0
makeblastdb -in ../Ghirsutum_genome.fasta -dbtype nucl -out Gh -parse_seqids




24.43–24.62


```shell
makeblastdb -in Ghirsutum_genome.fasta -dbtype nucl -out Gh -parse_seqids
```



## 位点聚合
~/cotton_fiber/gwas/combine_plot
获取位点的基因型数据


FS1 FS2的坐标根据附件中的基因获取

~/work/combine_plot #plot #ggplot2
- [x] 样本用376

~/cotton_fiber/genome/ng_introg_paper/TM-1


~/cotton_fiber/gwas/combine_plot


FS三个队列都试了，就是跑不了。但UHML能跑
ecaviar_run.sh: line 9: 451965 Segmentation fault      (core dumped)

bsub -q normal -o %J.out -e %J.err "sh ecaviar_run.sh FS 4DPA"
bsub -q normal -o %J.out -e %J.err "sh ecaviar_run.sh FS 8DPA"
bsub -q normal -o %J.out -e %J.err "sh ecaviar_run.sh FS 12DPA"
bsub -q normal -o %J.out -e %J.err "sh ecaviar_run.sh FS 16DPA"
bsub -q normal -o %J.out -e %J.err "sh ecaviar_run.sh FS 20DPA"


- [ ] D11位点做LD分析





# 2021-09-25 Saturday

## reading
http://zarlab.cs.ucla.edu/tag/ecaviar/





## eCAVIAR 

~/ga/ecaviar_plink_r2/phe1/ecaviar_input/FS/16DPA

是LD文件的问题，还是软件的bug
==Segmentation fault (core dumped)==


~/ga/ecaviar_plink_r2/phe1/ecaviar_input/UHML/16DPA/Garb_08G007740.LD

module load CAVIAR/2.2

eCAVIAR -l Garb_08G007740.LD -z Garb_08G007740.GWAS.Z -l Garb_08G007740.LD -z Garb_08G007740.EQTL.Z -f 1 -c 2 -o Garb_08G007740


gene=Garb_13G025020
eCAVIAR -l $gene.LD -z $gene.GWAS.Z -l $gene.LD -z $gene.EQTL.Z -f 1 -c 2 -o $gene

测试路径：
/public/home/zyqi/ga/ecaviar_plink_r2/phe1/ecaviar_input/UHML/12DPA
/public/home/zyqi/ga/ecaviar_plink_r2/phe1/ecaviar_input/FS/12DPA


## 软件安装
conda install -c bioconda r-coloc

FUSION 有点搞不清楚
为什么先建了所有的目录，难道这目录不是我建的？


## hotspot
~/work/ga/hotspot_all/plot2
/data/cotton/zyqi/ga_hotspot

/data/cotton/zyqi/ga_hotspot

cis eQTL不是用的我重新跑的


- [ ] 过滤掉一些基因，根据R2和表达量
- [ ] 合并hotspot，根据距离和LD？
- [ ] 动态性，有多少基因是重合的？



## TWAS
~/ga/fusion_1 
phe1，同时分析了 coloc


fusion 计算权重
四倍体：7000多基因能剩下5000多
二倍体：7000多只能剩下2000多？？？

重新计算权重看看
~/ga/fusion_new_compute_weight/
明天提交，今天节点不够

## ecaviar
- [ ] ecaviar 500k 还没做
- [ ] 为什么LD文件会使软件报错





# 2021-09-26 Sunday

 

 optparse

投任务有点困难，节点够，但我只能投上去177个任务，有的还SSUSP
是不是应该晚上十点以后投。


~/cotton_fiber/ecaviar/make_ecaviar_input_file/data/376_use_all_eGene_fillavg
将原来的UHML删除，UHML_1改成UHML

`将FS/*D/data 删掉好了`


清理文件是以后一个重要的任务

我清理了这个目录下的`SRR*`目录，`diskquota`竟然没有变化，是不是还需要等待
/data/cotton/zyqi/download


bss "mv ~/ga/fusion /data/cotton/zyqi/Ga/Ga_fusion"

## 清理硬盘
```
(mashr) 14:43 zyqi@mn02:~/cotton_fiber
$bss -q high "mv fusion ecaviar /data/cotton/zyqi/archives"
Job <43031365> is submitted to queue <high>.
  
```

二倍体FUSION路径
/data/cotton/zyqi/Ga/Ga_fusion

## 任务
每个位点的LD



module load ImageMagick/7.0.8-63
module load plink/1.9
ShowLDSVG -InPreFix chr05_FL_phe1 -OutPut  chr05_FL_phe1_modify -InGWAS UHML_gwas_phe1.txt -Cutline 5 - ShowNum -PointSize 3


## 明日计划
- [ ] 所有位点的LD plot，~/ga/LDblock
- [x] TWAS 结果过滤，所有位点
- [ ] FUSION脚本计算R2
- [ ] coloc原理，PP3，PP4
- [ ] mv，/data/cotton/zyqi/archives
- [ ] TWAS每列的含义
- [ ] 找出block区域







# 2021-09-27 Monday

## LDBlockShow

~/ga/LDblock



## 表达量与性状的关系

表达量 log
/data/cotton/zyqi/Ga/Ga_fusion/fusion/expression/8DPA_FPKM_log.txt

性状 phe1
/data/cotton/TMP/LiJianying_WildPAN/Ga_GWAS/Phe1/Phe1_filter2.txt


基因型
~/ga/tensorQTL/Genotypes



phe1, UHML
Garb_07G017500, 4D,8D
Garb_07G017510, 8D



phe1, UHML
Garb_05G038130, 8D
Garb_05G037970, 8D

## 基因型与性状的关系


# 2021-09-29 Wednesday

fusion重新计算weight。看看weights文件中留下的基因是不是还是只有2000多个。
/data/cotton/zyqi/Ga/Ga_fusion/fusion_new_compute_weight


## locus plot
~/ga/ecaviar/locus_plot

python locus_plot_argv.py 4DPA FS Garb_11G016980 Chr11_19272651 200000 /public/home/zyqi/ga/ecaviar/phe2/locus_plot_input wang_select_FS


## 得到locus plot input
~/ga/ecaviar/locus_plot/new_locus_0929



coloc是怎么计算LD的？

## FUSION
gcta_nr_robust要在环境变量中 #FUSION 
![[Pasted image 20210929145027.png]]



## 调控网络
用新的cis重新做分析
~/ga/hotspot/hot_scan_newCis/hotspot_all/tracks_c

- [x] 找出每个hotspot调控的基因





# 2021-09-30 Thursday


## 绘图脚本

~/ga/expression_trait/expression_plot.ipynb 
位点表达量小提琴图 #plot #ggplot2

~/ga/ecaviar/locus_plot/new_locus_0929/locus_plot.ipynb
locus plot #plot  #python 

~/ga/mashr_tensorQTL/effect_corr/barplot.ipynb
表达量和cis-eQTL效应的相关性 #plot #python


~/ga/annovar/function_enrichment/plot.ipynb
#plot #python

柱形图
~/ga/tensorQTL/independent_plot.ipynb
#plot #ggplot2

位点聚合
~/work/combine_plot/nofilter.ipynb
#plot 


QTL位点
- [ ] 可以加FS11 phe2




~/ga/LDblock





#obsidian 日历中的小黑点跟代码块有关？
跟字数有关。250字，https://github.com/liamcain/obsidian-calendar-plugin




