# 2021-10-01 Friday

/data/cotton/zyqi/Ga/Ga_fusion/fusion_new_compute_weight
投任务，要在twas环境 #TWAS 


回复审稿意见





# 2021-10-04 Monday

communications biology 审稿意见回复



polygenic scores  https://en.wikipedia.org/wiki/Polygenic_score
In genetics, a polygenic score (PGS), also called a polygenic risk score (PRS), genetic risk score, or genome-wide score, is a number that summarises the estimated effect of many genetic variants on an individual's phenotype, ==typically calculated as a weighted sum of trait-associated alleles==.[1][2][3] It reflects an individual's estimated genetic predisposition for a given trait and can be used as a predictor for that trait.[4][5][6][7][8] In other words, it gives an estimate of how likely an individual is to have a given trait only based on genetics, without taking environmental factors into account. Polygenic scores are widely used in animal breeding and plant breeding (usually termed genomic prediction or genomic selection) due to their efficacy in improving livestock breeding and crops.[9] In humans, polygenic scores are typically generated from genome-wide association study (GWAS) data.

是一个总结许多遗传变异对个体表型的估计影响的数字，通常计算为性状相关等位基因的加权总和
和我的位点的聚合是差不多的意思。==他们怎么计算这个得分？==







# 2021-10-08 Friday

The Zen of Python, by Tim Peters

Beautiful is better than ugly.
Explicit is better than implicit.
Simple is better than complex.
Complex is better than complicated.
Flat is better than nested.
Sparse is better than dense.
Readability counts.
Special cases aren't special enough to break the rules.
Although practicality beats purity.
Errors should never pass silently.
Unless explicitly silenced.
In the face of ambiguity, refuse the temptation to guess.
There should be one-- and preferably only one --obvious way to do it.
Although that way may not be obvious at first unless you're Dutch.
Now is better than never.
Although never is often better than *right* now.
If the implementation is hard to explain, it's a bad idea.
If the implementation is easy to explain, it may be a good idea.
Namespaces are one honking great idea -- let's do more of those!




做优于不做，
然而不假思索还不如不做。

很难解释的，必然是坏方法。
很好解释的，可能是好方法。





meta 分析，医学上用的比较多，荟萃分析
http://blog.sciencenet.cn/blog-41174-1004329.html
https://bookdown.org/yufree/sciguide/lib.html#%E8%8D%9F%E8%90%83%E5%88%86%E6%9E%90





# 2021-10-09 Saturday

反式eQTL
![[Pasted image 20211009213821.png]]

网络图
![[Pasted image 20211009213943.png]]





fisher 精确检验置信区间的计算问题，参见CV上的讨论。
https://stats.stackexchange.com/questions/409219/how-does-fisher-test-calculate-the-confidence-interval-for-the-odds-ratio-in-r





# 2021-10-10 Sunday



fusion 跑完
eQTL或者基因在gwas区间内
~/Downloads/TWAS_FDRsig_mergeGWAS.txt


gwas LD
~/ga/LDblock_new


fisher精确检验

gitbook
公链没有同步。
还是把git用熟



# 2021-10-11 Monday

08:58 用sz传new_modify_png.zip导致我电脑卡死




14:51 3000多个SNP跑LD block图一天都没跑完

 

 

 


 - [ ] coloc怎么确定causal SNP 

 


 chrome 黑色主题
 google 深色主题


 ~/cotton_fiber/genome/Garboreum_genome_HAU_v1.0/change.Chr_genome_final_gene_1based.bed





# 2021-10-12 Tuesday

locus plot
/data/cotton/zyqi/Ga/locus_plot_1011


小提琴图
~/ga/expression_trait_new




## 所有性状GWAS
gwas 五个性状都跑一下，跑完了
/data/cotton/zyqi/Ga/Ga_GWAS_new/mean
/data/cotton/zyqi/Ga/Ga_GWAS_new 有备份





# 2021-10-13 Wednesday

read -a PHENO_NAME < header.txt


根据LD block确定位点范围，然后找基因，并注释

重叠的算两个，chr05有的怪。

~/ga/LDblock_new/QTL_gene

~/ga/annovar


~/ga/LDblock_new/QTL_gene/nonsynonymous_gene_annot.txt

明天分析hotspot
8575
~/ga/hotspot/hot_scan_newCis/hotspot_all/tracks_c

- [x] 位点整理，非同义突变
- [x] chr5_hotspot调控的基因


85535945





# 2021-10-14 Thursday

rsync -e 'ssh -p 22333'  -avp hotspot_result_new lab:~/ga/hotspot/hot_scan_newCis


~/ga/hotspot/hot_scan_newCis/hot_chr05


- [x] chr05，90M的位点也考虑一下



明日计划，GO结果注释棉花基因信息









# 2021-10-15 Friday

~/ga/LDblock_new


​      

pp3很高，pp4很低代表什么？


chr05，hotspot动态变化
~/ga/hotspot/hot_scan_newCis/hotspot_all/tracks_c



tensorQTL independent 增加pos
~/ga/tensorQTL/independent/number



- [ ] fusion 考虑了群体结构啥的吗？它的最显著的eQTL是tensorQTL最显著的eQTL吗？





# 2021-10-16 Saturday

- [ ] jupyter lab 表格左对齐jupyterlab的cell？ #失败


_Z_ score of association between gene and schizophrenia plotted on the _y_ axis


/data/cotton/zyqi/Ga/Ga_fusion/coloc/mean


`> IRkernel::installspec(name='ir_twas',displayname='R (twas)')` #jupyterlab 


环境：mashr
> install.packages("coloc")





# 2021-10-17 Sunday

反式eQTL circos图

~/work/RIdeogram


我发现表达数据预处理的方式不一样
tensorQTL 用的qqnorm

fusion用的log


plink r 有负值
/data/cotton/zyqi/Ga/coloc_tensorQTL/test_plink_r #plink

获得coloc输入数据
/data/cotton/zyqi/Ga/coloc_tensorQTL #coloc

因为只假设1个causal SNP，所以不用LD


/data/cotton/zyqi/Ga/Ga_fusion/coloc/mean


~/ga/expression_trait_new_causalSNP





# 2021-10-18 Monday

cp -d 复制软链接而不是源文件


~/ga/coloc_tensorQTL/locus_plot_causalSNP

coloc 和 locus plot 一起





fusion 与gwas 取交集
/data/cotton/zyqi/Ga/Ga_fusion/fusion_new_result_gwas



挑选出来的QTL（为按LD筛选）
~/ga/LDblock_new/QTL_select_16.txt

按LD筛选之后的QTL
~/ga/LDblock_new/QTL_gene_FL5_3/qtl.bed





# 2021-10-21 Thursday





一些circos路径 #circos
~/work/circos/plot2
~/Lab/upland_magic/model/201228/circos_all/
~/work/ga/hotspot_all/hotspot_result_new
/Users/zhengyangqi/work/trash_0809/back_0527/FigureYa14circos





# 2021-10-22 Friday

拟南芥ID背景
grep 筛选文件，顺序跟原文件一样，而不是跟筛选条件一样 #grep
~/Desktop/gwas位点分析1013/GO
用diff可以证明
diff chr5_hotspot_gene_remain_select.txt chr5_hotspot_gene_remain.txt





# 2021-10-23 Saturday

#R dplyr分组操作，会按字母排序





# 2021-10-24 Sunday

/Users/zhengyangqi/Desktop/gwas位点分析1013/enrichplot

enrichplot的源代码



~/Desktop/gwas位点分析1013/GO/plot





# 2021-10-26 Tuesday

15:25
小提琴图+箱线图。~/ga/coloc_tensorQTL/expression_trait_new_causalSNP
AI 编辑





# 2021-10-27 Wednesday

生成基因+promote的坐标





# 2021-10-28 Thursday

~/work/ga/circos/data_prepare/share_specific/test





# 2021-10-29 Friday

hotspot基因过滤
~/ga/coloc_tensorQTL/expression_trait_new_causalSNP_qqnorm/trans-Zscore


过滤标准
