提取启动子序列，在尹师兄脚本的基础上改，py2->py3

```python
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

'''
 @Author:       qizhengyang <qizhengyang17@gmail.com>
 @DateTime:     2020年04月29日14:39:43
 @Description:  查找启动子3000
'''


def rc(seq):
    return seq[::-1].translate(str.maketrans('ACGT', 'TGCA'))


def read_fasta(fastafile):
    ids = []
    seqs = []
    f = open(fastafile)
    seq = ''
    for line in f:
        line = line.strip()
        if line.startswith('>'):
            ids.append(line[1:])
            if seq:
                seqs.append(seq)
                seq = ''
        else:
            seq += line.upper()
    seqs.append(seq)
    return ids, seqs


def read_query(query):
    query_ids = []
    with open(query) as lines:
        for line in lines:
            line = line.strip()
            if line and line != '0':
                query_ids.append(line)
    return query_ids


def read_gene(gff):
    gene_position = {}
    with open(gff) as lines:
        for line in lines:
            if line.startswith('#'):
                pass
            else:
                if line.strip():
                    t = line.strip().split('\t')
                    ch = t[0]
                    start = int(t[3]) - 1
                    end = int(t[4]) - 1
                    strand = t[6]

                    if t[2] == 'gene':
                        # 这里根据不同的gff格式修改
                        gene = t[8].split(';')[0].replace('ID=gene:', '')
                        # gene = gene.split('.')[0]
                        gene_position[gene] = [ch, start, end, strand]
    return gene_position


query = './SUNid.txt'
genome = './S_lycopersicum_chromosomes.4.00.fa'
gff = './ITAG4.0_gene_models.gff'
outfile = './SUN_promoter_20200505.fa'


f = open(outfile, 'w')
left = 3000
right = 0
gene_position = read_gene(gff)
ids, seqs = read_fasta(genome)
query_ids = read_query(query)

for id in query_ids:
    t = gene_position.get(id, [])
    ch = t[0]
    print(t, id)
    start = t[1]
    end = t[2]
    strand = t[3]
    for i, gid in enumerate(ids):
        if gid == ch:
            if strand == '+':
                n_start = start - left
                n_end = start
                # print gid, strand
                cseq = seqs[i][n_start:n_end]
                # assert len(cseq) == left + right
                # f.write('>%s %s\n%s\n'%(id, strand, cseq))
                f.write('>%s\n%s\n' % (id, cseq))
            else:
                n_start = end + 1
                n_end = end + left + 1
                # print gid, strand
                cseq = rc(seqs[i][n_start:n_end])
                # assert len(cseq) == left + right
                # f.write('>%s %s\n%s\n'%(id, strand, cseq))
                f.write('>%s\n%s\n' % (id, cseq))
            break
f.close()

```

